lib/core/maintenance_mode.rb

```diff
@@ -84,9 +84,13 @@ def start_or_stop_maintenance_workload(action)
   def switch_domain_workload(to:)
     step("Switching workload for domain '#{domain_data['name']}' to '#{to}'") do
       cp.set_domain_workload(domain_data, to)
+    end
+
+    step("Waiting for changes to take effect", retry_on_failure: true, wait: 10, max_retry_count: 3) do
+      refetched_domain_data = cp.fetch_domain(domain_data["name"])
+      raise "Can't find domain" if refetched_domain_data.nil?
 
-      # Give it a bit of time for the domain to update
-      Kernel.sleep(30)
+      cp.domain_workload_matches?(refetched_domain_data, to)
```

Comment:
@rafaelgomesxyz Is this check enough to be sure that domain was updated?

Reply:
> So we'd need a different method for checking it 

Do you have any ideas how can we check this? Maybe controlplane displays some message while workload change takes effect?

Reply:
@zzaakiirr A way I can think of is to make a real request to the domain and see if it's displaying the regular page or the maintenance page.

I'm not sure if there's a better way to do it.